packages/react-native/android/upload-sourcemaps.gradle

```diff
@@ -0,0 +1,48 @@
+import groovy.json.JsonBuilder
+import groovy.json.JsonSlurper
+
+
+tasks.register("uploadSourceMapsToBacktrace") {
+    group = "backtrace"
+    description = "This task uploads source maps generated by the react-native builder to Backtrace based on the .backtracejsrc file."
+
+    doLast {
+        // Fetch the build variant (debug or release)
+        // Default to 'release' if not set
+        def buildVariant = project.hasProperty("buildVariant") ? project.buildVariant.toLowerCase() : "release"
+
+        def debugIdPathEnv = System.getenv("DEBUG_ID_PATH")
+
+        def debugIdFile = debugIdPathEnv ? file(debugIdPathEnv) : layout.buildDirectory
+                .dir("intermediates/sourcemaps/react/${buildVariant}/.backtrace-sourcemap-id")
+                .get()
+                .asFile
+
+        if (!debugIdFile.exists()) {
+            println("Backtrace: Cannot find .backtrace-sourcemap-id debug info.")
+            return
+        }
+
+        def sourcemapDestinationDirectory = layout.buildDirectory.dir("generated/sourcemaps/react").get().asFile
+        def mapFiles = fileTree(dir: sourcemapDestinationDirectory, include: '**/*.map').files
+        def mapFile = mapFiles ? mapFiles.iterator().next() : null
+
+        if (!mapFile) {
+            println("Backtrace: Cannot find final source map file.")
+            return
+        }
+
+        def jsonSlurper = new JsonSlurper()
+        def mapData = jsonSlurper.parse(mapFile)
+        mapData.debugId = debugIdFile.text
+        def jsonBuilder = new JsonBuilder(mapData)
+        mapFile.text = jsonBuilder.toPrettyString()
```

Comment:
not sure what was the previous serialization, but is pretty text required here? It will increase the map size

Reply:
What I mean is that you can omit the formatting, i.e. pretty string would be this:
```
{
  "a": 123
}
```
while non-pretty (what we want to save size):
```
{"a": 123}
```

Reply:
It's changed to a default serialization option